stub/src/main/java/io/grpc/stub/ServerCallStreamObserver.java

```diff
@@ -143,4 +145,28 @@ public void disableAutoRequest() {
    */
   @Override
   public abstract void setMessageCompression(boolean enable);
+
+  /**
+   * Sets a {@link Runnable} to be executed when the call is closed correctly from the server's
+   * point of view: either {@link #onCompleted()} or {@link #onError(Throwable)} has been called,
+   * all the messages and trailing metadata have been put on the wire and the stream has been
+   * closed.
+   * Note however that the client still may have not received all the messages due to race
```

Comment:
"race conditions" makes it sound like it is part of the implementation. How about "due to network delay, client crashes, and cancellation races."

Combine this with the previous line?

Reply:
done